.github/workflows/_reusable_run_tests.yml

```diff
@@ -119,7 +118,7 @@ jobs:
         if: always()
         uses: actions/upload-artifact@v4
         with:
-          name: xcresults
+          name: XCResults for ${{ env.SANITIZED_BRANCH }} (${{ github.run_id }})
```

Comment:
`run_id` takes account of workflow run but not run attempt. Rerunning a job (for example if the first run attempt failed) will overwrite the this. I think we should **additionally** add the workflow run attempt to the name - `github.run_attempt`

Reply:
fixed